lib/commons/aria/get-unsupported-role-attrs.js

```diff
@@ -0,0 +1,60 @@
+import { getRole } from '../../commons/aria';
+import standards from '../../standards';
+import { querySelectorAllFilter } from '../../core/utils';
+
+/**
+ * Check that an element does not use any unsupported ARIA attributes for it's particular role.
+ * @param {VirtualNode} vNode
+ * @return {String[]}
+ */
+function getUnsupportedRoleAttrs(vNode) {
+  const unsupported = [];
+  const role = getRole(vNode);
+
+  if (!role) {
+    return unsupported;
+  }
+  const attrs = vNode.attrNames;
+  const unsupportedAttrs = standards.ariaRoles[role].unsupportedAttrs;
+
+  if (!unsupportedAttrs) {
+    return unsupported;
+  }
+
+  const preChecks = {
+    // aria-required is not supported on radiogroup but only
+    // if set to true and no other descendant form element is
+    // required
+    // @see https://github.com/dequelabs/axe-core/pull/2767#discussion_r561272090
+    'aria-required': () => {
+      const requiredInputSelector =
+        ':is(input, textarea):is([aria-required="true"],[required])';
+      const requiredDescendants = querySelectorAllFilter(
+        vNode,
+        requiredInputSelector
+      ).length;
+
+      if (
+        role === 'radiogroup' &&
+        vNode.attr('aria-required') === 'true' &&
+        !requiredDescendants
+      ) {
+        return true;
+      }
+    }
```

Comment:
This seems like it should be its own function somewhere, not embedded into this one. Sticking this here interrupts the reading flow. Maybe just put this below his function?

Reply:
This is in the same code style as https://github.com/dequelabs/axe-core/blob/develop/lib/checks/aria/aria-valid-attr-value-evaluate.js#L39. Should we add a tech debt ticket to update that one?